middleware/faq.rst

```diff
@@ -541,6 +541,35 @@ In most cases, a complex ``--data-query`` argument is preferable to even one inp
+
+When one quantum in a quantum graph fails (e.g. by raising an exception), executors typically block all downstream quanta, even those that might be able to proceed with the outputs of the failed quantum.
+It's also possible for a quantum to nominally succeed while doing the wrong thing, which would allow downstream quanta to run but might also cause them to do the wrong thing.
+
+These situations are usually best addressed by building a new quantum graph that includes only the quanta that were affected by the problem.
+In some cases it may be possible to run this with the same output `~lsst.daf.butler.CollectionType.RUN` collection, but it is always possible and less error-prone to just create a new output `~lsst.daf.butler.CollectionType.RUN` in the same `~lsst.daf.butler.CollectionType.CHAINED` collection (which is what ``pipetask`` and ``bps`` do by default when an output collection is reused).
+The primary tools here are the ``--skip-existing-in`` and ``--select-tasks`` options to ``pipetask qgraph`` (or equivalently, the BPS ``extraQgraphArgs`` setting).
+The former is given a collection that is queried for existing outputs, and any quanta that already have a metadata dataset present in that collection are considered done and dropped from the new graph.
+The latter takes a :ref:`pipeline graph subset expression <pipeline-graph-subset-expressions>` that can be used to easily identify the tasks that were affected by a failure.
+
+There are a few different scenarios to consider:
+
+- If a task with label ``unfixable`` fails in one or more quanta, and you just want to move on without its outputs, modify the original quantum-graph build command with ``--skip-existing-in <output> --select-tasks '> unfixable'``.
+  This will only attempt to run tasks that are strictly downstream of ``unfixable``, and it will skip any quanta that already ran (i.e. because they didn't have upstream failures).
+
+- If a task with label ``fixable`` fails in one more quanta, but you have fixed the problem and now expect it to succeed at least some of the time, use ``--skip-existing-in <output> --select-tasks '>= fixable'``.
+  This will run both the problematic task and everything downstream of it, again skipping any quanta that already ran successfully.
+  In this case ``-skip-existing-in <output>`` is actually all you need for correctness, but using ``--select-tasks`` should speed up the quantum graph build.
+
+- If a task with label ``regrettable`` doesn't actually fail, but it's produced bad outputs (or an improvement is available that you want to try), use ``--select-tasks '>= regrettable'`` to re-run that task and all downstream tasks regardless of whether they succeeded before.
+  This is also the right approach to take with quanta that incorrectly raised `NoWorkFound` or otherwise wrote no outputs without formally failing and blocking downstream quanta; these are considered "successes with caveats", not failures.
```

Comment:
I don't know if readers would find hints about difference helpful (e.g., note the equal sign, note that --skip-existing-in is not here.  It may just be because of reading it in the raw text with weird highlighting.

Reply:
I've switched to sub-heading instead of bullet points and moved the command-line options onto full-line code blocks, which should draw attention to the differences.